sap/rest/gcts/simple.py

```diff
@@ -90,8 +98,65 @@ def clone(connection, url, rid, vsid='6IT', start_dir='src/', vcs_token=None, er
     if repo.is_cloned:
         _mod_log().info('Not cloning the repository "%s": already performed', repo.rid)
         return repo
+    with (
+        abap_modifications_disabled(repo, progress_consumer) if no_import else context_stub(),
+        buffer_only_enabled(repo, progress_consumer) if buffer_only else context_stub()
+    ):
+        repo.clone()
+    return repo
+
+
+# pylint: disable=too-many-locals
+def clone_with_task(connection, url, rid, vsid='6IT', start_dir='src/', vcs_token=None,
+                    error_exists=True, role='SOURCE', typ='GITHUB', no_import=False,
+                    buffer_only=False, wait_for_ready: int = 600, poll_period: int = 30,
+                    progress_consumer: LogTaskOperationProgress | None = None) -> Repository:
+    """Creates and clones the repository in the target systems and schedules a cloning task"""
+
+    repo = create(
+        connection=connection,
+        url=url,
+        rid=rid,
+        vsid=vsid,
+        start_dir=start_dir,
+        vcs_token=vcs_token,
+        error_exists=error_exists,
+        role=role,
+        typ=typ,
+    )
+    if progress_consumer:
+        progress_consumer.progress_message(f'Repository "{repo.rid}" has been created.')
+
+    if repo.is_cloned:
+        if progress_consumer:
+            progress_consumer.progress_message(f'Not cloning the repository "{repo.rid}": already performed.')
+        return repo
+    with (
+        abap_modifications_disabled(repo, progress_consumer) if no_import else context_stub(),
+        buffer_only_enabled(repo, progress_consumer) if buffer_only else context_stub()
```

Comment:
The context managers are supposed to temporarily modify repository configuration. In the case of a network error or something like that, the code execution is interrupted and the configuration changes will become persistent. To avoid user confusion, the program should let the end user know how to revert the configuration changes. Look at the example at: https://github.com/jfilak/sapcli/blob/master/sap/cli/gcts.py#L968